src/op_analytics/cli/subcommands/pulls/defillama/protocols.py

```diff
@@ -187,17 +192,21 @@ def extract_protocol_tvl_to_dataframes(protocol_data: dict) -> pl.DataFrame:
             tokens_entries = chain_data.get("tokensInUsd", [])
             for tokens_entry in tokens_entries:
                 dateval = dt_fromepoch(tokens_entry["date"])
+                timeval = datetime_fromepoch(tokens_entry["date"]).time()
 
                 if date_fromstr(dateval) < TVL_TABLE_CUTOFF_DATE:
                     continue
 
+                if timeval != datetime.min.time():
```

Comment:
for this check, could you create a utility in the `time.py` module that checks if the epoch is a date:
```
def epoch_is_date(epoch: int) -> bool:
    """Return true if seconds since epoch is at exactly midnight."""
    return epoch % 86400 == 0
```

then here you can do 
```
if not epoch_is_date(tokens_entry["date"]): 
    continue
```